controllers/argocd/prometheus.go

```diff
@@ -247,3 +248,150 @@ func (r *ReconcileArgoCD) reconcileServerMetricsServiceMonitor(cr *argoprojv1a1.
 	}
 	return r.Client.Create(context.TODO(), sm)
 }
+
+// reconcilePrometheusRule reconciles the PrometheusRule that triggers alerts based on workload statuses
+func (r *ReconcileArgoCD) reconcilePrometheusRule(cr *argoprojv1a1.ArgoCD) error {
+
+	promRule := newPrometheusRule(cr.Namespace, "argocd-component-status-alert")
+
+	if argoutil.IsObjectFound(r.Client, cr.Namespace, promRule.Name, promRule) {
+
+		if !cr.Spec.Monitoring.Enabled {
+			// PrometheusRule exists but enabled flag has been set to false, delete the PrometheusRule
+			log.Info("instance monitoring disabled, deleting component status tracking prometheusRule")
+			return r.Client.Delete(context.TODO(), promRule)
+		}
+		return nil // PrometheusRule found, do nothing
```

Comment:
the code just returns nil here
how would we know for sure that this is actually the branch that is being executed? I think it would be hard to tell if it went into a different branch that also returns nil

Reply:
As long as this scenario returns `nil` it's good enough:

1. Create the prometheu rule `promRule := newPrometheusRule(cr.Namespace, "argocd-component-status-alert")` and make sure it's present in the cluster.
2. Set the enabled flag to false.
3. Trigger a reconcile.

Reply:
updated